### ISSUE TITLE
maximum call stack size at friends.js line 1137

### ISSUE THREAD

**Issue by JustAzul** [DoctorMcKay/node-steam-user#345]
## Describe the bug
maximum call stack size

## Versions
v4.19.3

## Screenshots and Error Logs
![unknown](https://user-images.githubusercontent.com/30638036/110256855-e71add80-7f79-11eb-83be-823db7df354c.png)

**Comment by DoctorMcKay** [comment]
I haven't been able to reproduce this.

**Comment by JustAzul** [comment]
this seens to happens when the logged in account has a lot of friends, and this makes the event 'friendsList' not to be triggered, causing the property 'myFriends' to be empty

**Comment by JustAzul** [comment]
actualy the friendsList event doesnt seen to be triggered at all, not sure why, 
![code](https://user-images.githubusercontent.com/30638036/116009523-99713780-a5f0-11eb-91bf-ed55f405665c.JPG)
![image](https://user-images.githubusercontent.com/30638036/116009519-91b19300-a5f0-11eb-8b2a-2d69b785c50e.png)

**Comment by DoctorMcKay** [comment]
Are you saying that `friendsList` is never triggered, or that it's not triggered when you get the error? Or do you get the error every time you log on?

**Comment by JustAzul** [comment]
`friendsList` isn't being triggered, but im not sure anymore whats causing it, at first i thought it was the `maximum call stack error`, but on my tests it wasn't being triggered even without `call stack error`, but the weird thing is, at first log on, it was triggering, but on next ones, it wasn't, and i've checked the myFriends property, it was empty making the script unable to manage friend list

so, something is making the `friendsList` event unable to be triggered(also making the property myFriends empty), but the script seemed to be fetching users details(user event was being triggered)..

do you have any tests in mind so i can help to fix that issue?

to be more specific:
1. the account used does not have any sort of limitations in steam, and works just fine, no trade holds, bans, or whatever limitations steam have but market usage, the account does not use the steam market.
2. more users seems to be experiencing this issue.
3. a specific user is having the maximum call stack error, and after remove some friends, the error disappears.
4. its possible that more users is having the `maximum call stack error`, since only one have reported it, but others have similar issue.

**Comment by DoctorMcKay** [comment]
If I could get debug output from you, that would be helpful.

```js
Client.on('debug', (msg) => {
   console.log(msg); // or whatever custom debug log function you want to use
});

**Comment by JustAzul** [comment]
![image](https://user-images.githubusercontent.com/30638036/116235284-0b05ce80-a734-11eb-8720-fdf59bef4f5b.png)

![image](https://user-images.githubusercontent.com/30638036/116235149-df82e400-a733-11eb-8f8f-d5521160c3d1.png)

this `Got incomplete message; expecting 4864 more bytes` seems to be the problem

**Comment by JustAzul** [comment]
if anyone else is having this issue, a failover to https://developer.valvesoftware.com/wiki/Steam_Web_API#GetFriendList_.28v0001.29 its a good idea.

**Comment by DoctorMcKay** [comment]
> this Got incomplete message; expecting 4864 more bytes doesn't sound harmless

It is, that's normal.

I really don't know why you wouldn't be receiving your friends list. It gets pushed down automatically by the CM; it's not something that can be requested. The same is true of friend persona states. I would expect to see a `ClientFriendsList` handled message line, but it just isn't there. Are you sure that's the entire debug output? There should be some more output around when the client connects and logs on.

**Comment by JustAzul** [comment]
> > this Got incomplete message; expecting 4864 more bytes doesn't sound harmless
> 
> It is, that's normal.
> 
> I really don't know why you wouldn't be receiving your friends list. It gets pushed down automatically by the CM; it's not something that can be requested. The same is true of friend persona states. I would expect to see a `ClientFriendsList` handled message line, but it just isn't there. Are you sure that's the entire debug output? There should be some more output around when the client connects and logs on.

yeah, that's everything. some accounts just doesn't download the friend list somehow, and i have no idea why

**Comment by thecatontheflat** [comment]
I get this error sporadically too, but on a different line:

```
Exception in PromiseRejectCallback:
steam-user/components/friends.js:1190
			return resolve(user);
```
